clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/validators/BasicAmazonDeployDescriptionValidator.groovy

```diff
@@ -73,6 +74,27 @@ class BasicAmazonDeployDescriptionValidator extends AmazonDescriptionValidationS
     if (!description.source?.useSourceCapacity) {
       validateCapacity description, errors
     }
+
+    // certain features are only supported when AWS EC2 Launch Templates support is enabled
+    if (hasLaunchTemplateFeature(description) && !description.setLaunchTemplate) {
+      errors.rejectValue "setLaunchTemplate", "basicAmazonDeployDescription.setLaunchTemplate.not.enabled"
+    }
+
+    // unlimitedCpuCredits (set to true / false) is valid only with instance families t2/t3/t3a
+    if (description.unlimitedCpuCredits != null && !InstanceTypeUtils.isBurstingSupported(description.instanceType)) {
+      errors.rejectValue "unlimitedCpuCredits", "basicAmazonDeployDescription.bursting.not.supported.by.instanceType"
+    }
+  }
+
+  /**
+   * Check if the description includes a launch template feature
+   * @param description
+   * @return true if the description includes one or more features supported by launch template only.
+   */
+  private boolean hasLaunchTemplateFeature(BasicAmazonDeployDescription description) {
```

Comment:
_Highlighting a potential area of interest:)_

Reply:
Discussed with @caseyhebebrand offline and changed the impl from error to [warnings](https://github.com/spinnaker/clouddriver/blob/847313664d814f405ab454c78d7b397372fd5c3f/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/validators/BasicAmazonDeployDescriptionValidator.groovy#L85).